### PR TITLE
Add new structure for OrgVdcRecord and additional attributes for QueryResultVappTemplateType

### PR DESCRIPTION
**Pull request by whiterm**
The proposed commit adds a QueryResultOrgVdcRecordType structure and additional fields for the QueryResultOrgVdcRecordType structure.

This Pull Request isn't associated with an Issue, because this is a very simple commit, it extends the model described in the schema

### CODE REVIEW

File: types/v56/types.go [hunk 1]
```diff
@@ -2179,29 +2180,44 @@ type QueryResultCatalogItemType struct {
...
+	IsEnabled              bool      `xml:"isEnabled,attr,omitempty"`          // True if the vApp template is enabled
+	Status                 string    `xml:"status,attr,omitempty"`             // Status
+	IsDeployed             bool      `xml:"isDeployed,attr,omitempty"`         // True if this entity is deployed
+	IsExpired              bool      `xml:"isExpired,attr,omitempty"`          // True if this entity is expired
+	StorageProfileName     string    `xml:"storageProfileName,attr,omitempty"` // Storage profile name
+	Version                string    `xml:"version,attr,omitempty"`            // Storage profile name
+	LastSuccessfulSync     string    `xml:"lastSuccessfulSync,attr,omitempty"` // Date of last successful sync
+	Link                   *Link     `xml:"Link,omitempty"`
+	Metadata               *Metadata `xml:"Metadata,omitempty"`
+	NumberOfShadowVMs      int       `xml:"numberOfShadowVMs,attr,omitempty"`
```
Comment (dataclouder) [inline types/v56/types.go:2206]: 1. Why are we adding these fields?
2. Are they supported in every version?
3. Have you run a test to make sure the addition don't break existing code?

File: types/v56/types.go [hunk 2]
```diff
@@ -2132,6 +2132,7 @@ type QueryResultRecordsType struct {
 	VMRecord                        []*QueryResultVMRecordType                        `xml:"VMRecord"`                        // A record representing a VM result.
 	AdminVMRecord                   []*QueryResultVMRecordType                        `xml:"AdminVMRecord"`                   // A record representing a Admin VM result.
 	VAppRecord                      []*QueryResultVAppRecordType                      `xml:"VAppRecord"`                      // A record representing a VApp result.
+	OrgVdcRecord                    []*QueryResultOrgVdcRecordType                    `xml:"OrgVdcRecord"`                    // A record representing VDC
```
Comment (dataclouder) [inline types/v56/types.go:2135]: What are we going to do with this field? No functions are using it.
Also, do we need an `AdminOrgVdcRecord` field?

File: types/v56/types.go [hunk 3]
```diff
@@ -2342,6 +2358,50 @@ type QueryResultVMWProviderVdcRecordType struct {
 	MemoryOverheadMB        int64  `xml:"memoryOverheadMB,attr,omitempty"`
 }
 
+// QueryResultOrgVdcRecordType represents all VDC in organization
+// Type: QueryResultOrgVdcRecordType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Since: 1.5
+type QueryResultOrgVdcRecordType struct {
```
Comment (dataclouder) [inline types/v56/types.go:2365]: We would need a test for the usage of this type

**Review by dataclouder — COMMENTED** [review]
Thanks for your contribution.
Please answer the comments in the code

File: types/v56/types.go (hunk 3 above)
Comment (whiterm) [reply types/v56/types.go:2365]: I added a method for usage this type and a test

File: types/v56/types.go (hunk 2 above)
Comment (whiterm) [reply types/v56/types.go:2135]: 1. I added a method for usage this type and a test.
2. I don't think so, I didn`t find  AdminOrgVdcRecord in the schema and docs https://code.vmware.com/apis/722/vmware-cloud-director

File: types/v56/types.go (hunk 1 above)
Comment (whiterm) [reply types/v56/types.go:2206]: I didn't find these fields in [docs] (https://code.vmware.com/apis/722/vmware-cloud-director/doc/doc/types/QueryResultVAppTemplateRecordType.html) but our vcd instance returns them. This is very strange and I don't know why.

**Comment by Didainius** [comment]
Sorry for this drag - we are catching up on open PRs - would you be able to merge with latest master? Apparently we had to implement `QueryOrgVdcList` function as well - maybe it fits your needs

**Pull request closed without merging**